zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/IdentitySetupInitializeProcessor.java

```diff
@@ -250,7 +254,17 @@ private void createMapping(
 
   private boolean assignEntityToRole(
       final long roleKey, final long entityKey, final EntityType entityType) {
-    final var isAlreadyAssigned = roleState.getEntityType(roleKey, entityKey).isPresent();
+
+    final var isAlreadyAssigned =
+        switch (entityType) {
+          case USER ->
+              membershipState.hasRelation(
```

Comment:
💭 I think that this could become a bit of a verbose switch in time (as more entities are added to the membership state), we could look to provide a series of helper/wrapper methods to reduce the external verbosity i.e. `membershipState.isUserAMemberOfRole(userId, roleId)` it doesn't achieve a significant benefit though so can be considered NABD.

Reply:
The verbosity is only temporary, as soon as all relationships are managed this way, we can make a single call to `membershipState.hasRelation(...)` and just pass in whatever entity type we have.

Reply:
Makes sense :)